commit 86bd812676d1e99968f395ad82ccaa2db7965748
Author: Hauke Mehrtens <hauke@hauke-m.de>
Date:   Thu Sep 19 22:04:04 2013 +0200

    MIPS: BCM47XX: print board name in machine entry in cpuinfo
    
    This will add the board name to the machine entry in /proc/cpuinfo.
    
    Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

--- a/arch/mips/bcm47xx/setup.c
+++ b/arch/mips/bcm47xx/setup.c
@@ -32,6 +32,7 @@
 #include <linux/ssb/ssb_embedded.h>
 #include <linux/bcma/bcma_soc.h>
 #include <asm/bootinfo.h>
+#include <asm/prom.h>
 #include <asm/reboot.h>
 #include <asm/time.h>
 #include <bcm47xx.h>
@@ -225,6 +226,7 @@ void __init plat_mem_setup(void)
 	_machine_halt = bcm47xx_machine_halt;
 	pm_power_off = bcm47xx_machine_halt;
 	bcm47xx_board_detect();
+	mips_set_machine_name(bcm47xx_board_get_name());
 }
 
 static int __init bcm47xx_register_bus_complete(void)
